app/components/animated_number/index.test.tsx

```diff
@@ -0,0 +1,176 @@
+
+        // @ts-expect-error mock implementation for testing
+        start: jest.fn().mockImplementation(() => a.setValue(b.toValue)),
+    }) as unknown as Animated.CompositeAnimation);
+
+    it('should render the non-animated number', () => {
+        const {getByTestId} = render(<AnimatedNumber animateToNumber={123}/>);
+
+        const text = getByTestId('no-animation-number');
+        expect(text.children[0]).toEqual('123');
+    });
+
+    it('should removed the non-animation number after getting the correct height', () => {
+        const {getByTestId, queryByTestId} = render(<AnimatedNumber animateToNumber={123}/>);
+
+        const text = getByTestId('no-animation-number');
+
+        fireEvent(text, 'onLayout', {nativeEvent: {layout: {height: NUMBER_HEIGHT}}});
+
+        const removedText = queryByTestId('no-animation-number');
+
+        expect(removedText).toBeNull();
+    });
+
+    it('should switch to the animated number view', async () => {
+        const {getByTestId} = render(<AnimatedNumber animateToNumber={123}/>);
+
+        const text = getByTestId('no-animation-number');
+
+        fireEvent(text, 'onLayout', {nativeEvent: {layout: {height: NUMBER_HEIGHT}}});
+
+        const animatedView = getByTestId('animation-number-main');
+        expect(animatedView).toBeTruthy();
+    });
+
+    describe.each([1, 23, 579, -123, 6789, 23456])('should show the correct number of animated views based on the digits', (animateToNumber: number) => {
+        it(`should display ${animateToNumber.toString().length} view(s) for ${animateToNumber}`, async () => {
+            const {getByTestId} = render(<AnimatedNumber animateToNumber={animateToNumber}/>);
+
+            const text = getByTestId('no-animation-number');
+
+            fireEvent(text, 'onLayout', {nativeEvent: {layout: {height: NUMBER_HEIGHT}}});
+
+            await waitFor(() => {
+                const animatedView = getByTestId('animation-number-main');
+                expect(animatedView.children).toHaveLength(animateToNumber.toString().length);
+            });
+        });
+    });
+
+    describe.each([123, 9982, 328, 1111, 2222, 3333])('should show the correct number', (animateToNumber: number) => {
+        it(`should display the number ${animateToNumber}`, async () => {
+            const {getByTestId} = render(<AnimatedNumber animateToNumber={animateToNumber}/>);
+
+            const text = getByTestId('no-animation-number');
+
+            fireEvent(text, 'onLayout', {nativeEvent: {layout: {height: NUMBER_HEIGHT}}});
+
+            const checkEachDigit = animateToNumber.toString().split('').map(async (number, index) => {
+                const useIndex = animateToNumber.toString().length - 1 - index;
+
+                let translateY = -1;
+
+                // every digit will have a row of 10 numbers, so the translateY should be the height of the number * the number * -1 (since the animation is going up)
+                await waitFor(() => {
```

Comment:
I don't think I need a waitFor() anymore since I'm using a Promise.all() afterwards to make sure all the numbers are correct.